src/main.c

```diff
@@ -16,14 +17,17 @@
 #include <rclc/executor.h>
 
 #include <rmw_microros/rmw_microros.h>
-//#include <microros_transports.h>
 
-#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc);for(;;){};}}
+#include "udp_transport.h"
+
+#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc);for(;;){k_msleep(100);};}}
```

Comment:
Why do we need 100 ms sleep here? Would you mind to enlighten me?

Reply:
Ah - for debugging of networking I was using CONFIG_NET_SHELL, which runs in another thread - this spinning at a higher priority stopped that from running.  The sleep yields to other threads